signxml/__init__.py

```diff
@@ -365,6 +367,22 @@ def _unpack(self, data, reference_uri):
                 raise InvalidInput("When using enveloped signature, **data** must be an XML element")
             c14n_input = self.get_root(data)
             doc_root = self.get_root(data)
+            if reference_uri is not None:
+                # Only sign the referenced element
+                if reference_uri.startswith('#'):
+                    reference_uri = reference_uri[1:]
+                targets = doc_root.xpath('//*[@Id="{0}"]'.format(reference_uri))
+                if not targets:
+                    targets = doc_root.xpath('//*[@ID="{0}"]'.format(reference_uri))
+                if len(targets) == 1:
+                    c14n_input = self.get_root(targets[0])
+                elif len(targets) == 0:
+                    raise InvalidInput("No reference found for reference_uri in enveloped signature")
+                else:
+                    raise InvalidInput("Multiple references found for reference_uri in enveloped signature")
+            else:
+                reference_uri = ""
+
```

Comment:
Would you mind refactoring this to use `XMLVerifier._resolve_reference()`? (that function will also require minor refactoring to accommodate this use case)